### PR TITLE
build: 4.0.1

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |y|
|New feature?   |n|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Do not save generated API in database log table

### CODE REVIEW

File: src/Common/Log/DoctrineHandler.php [hunk 1]
```diff
@@ -35,9 +38,28 @@ protected function write(array $record): void
...
 
         $this->logRepository->insertRecord($record);
     }
+
+    /**
+     * @param array<string, mixed> $context
+     *
+     * @return array<string, mixed>
+     */
+    private static function secertContext(array $context): array
```
Comment (theus77) [inline src/Common/Log/DoctrineHandler.php:54]:
```suggestion
    private static function secretContext(array $context): array
```

File: src/Common/Log/DoctrineHandler.php [hunk 2]
```diff
@@ -35,9 +38,28 @@ protected function write(array $record): void
         }
 
         $token = $this->tokenStorage->getToken();
-        $record['username'] = ($token instanceof TokenInterface ? $token->getUsername() : null);
-        $record['impersonator'] = ($token instanceof SwitchUserToken ? $token->getOriginalToken()->getUsername() : null);
+        $record['username'] = $token instanceof TokenInterface ? $token->getUsername() : null;
+        $record['impersonator'] = $token instanceof SwitchUserToken ? $token->getOriginalToken()->getUsername() : null;
+
+        $record['context'] = DoctrineHandler::secertContext($record['context']);
```
Comment (theus77) [inline src/Common/Log/DoctrineHandler.php:44]:
```suggestion
        $record['context'] = DoctrineHandler::secretContext($record['context']);
```

File: src/Common/Log/DoctrineHandler.php [hunk 3]
```diff
@@ -16,6 +16,9 @@ class DoctrineHandler extends AbstractProcessingHandler
     private TokenStorageInterface $tokenStorage;
     private int $minLevel;
 
+    private const SECRET_VALUE = '***';
+    private const SECRET_KEYS = ['api_key'];
```
Comment (theus77) [inline src/Common/Log/DoctrineHandler.php:20]: Define in the constructor? no? As it might be specific for the core then the emsch?

Comment (Davidmattei) [reply src/Common/Log/DoctrineHandler.php:20]: Would not do that for now. If we face it we can change it :)

**Pull request merged**